web3/providers/ipc.py

```diff
@@ -42,22 +42,52 @@ def get_default_ipc_path(testnet=False):
         testnet = ""
 
     if sys.platform == 'darwin':
-        return os.path.expanduser(os.path.join(
+        ipc_path = os.path.expanduser(os.path.join(
             "~",
             "Library",
             "Ethereum",
             testnet,
-            "geth.ipc",
+            "geth.ipc"
         ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
+        ipc_path = os.path.expanduser(os.path.join(
+            "~",
+            "Library",
+            "Application Support",
+            "io.parity.ethereum",
+            "jsonrpc.ipc"
+        ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
     elif sys.platform.startswith('linux'):
-        return os.path.expanduser(os.path.join(
+        ipc_path = os.path.expanduser(os.path.join(
             "~",
             ".ethereum",
             testnet,
-            "geth.ipc",
+            "geth.ipc"
         ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
+        ipc_path = os.path.expanduser(os.path.join(
+            "~",
+            ".local",
+            "share",
+            "io.parity.ethereum",
+            "jsonrpc.ipc"
+        ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
     elif sys.platform == 'win32':
-        return "\\\\.\\pipe\\geth.ipc"
+        if os.path.exists("\\\\.\\pipe\\geth.ipc"):
+            return "\\\\.\\pipe\\geth.ipc"
+        elif os.path.exists("\\\\.\\pipe\\jsonrpc.ipc"):
+            return "\\\\.\\pipe\\jsonrpc.ipc"
```

Comment:
A utility function like `first_available_path(paths)` could maybe remove some of the repetition here.